src/globus_sdk/authorizers/renewing.py

```diff
@@ -17,8 +17,12 @@
 # possible delays or clock skew.
 EXPIRES_ADJUST_SECONDS = 60
 
+# the type of the response which is produced by the authorizer, received by it, and
+# passed to the `on_refresh` callback
+ResponseT = t.TypeVar("ResponseT", bound="OAuthTokenResponse")
 
-class RenewingAuthorizer(GlobusAuthorizer, metaclass=abc.ABCMeta):
+
+class RenewingAuthorizer(GlobusAuthorizer, t.Generic[ResponseT], metaclass=abc.ABCMeta):
```

Comment:
What would happen to clients who have implemented their own custom RenewingAuthorizer if we add a new generic to the class signature?

Reply:
Implicitly, their subclasses become
```python
class MyAuthorizer(RenewingAuthorizer[Any]): ...
```

At runtime, they don't encounter any changes, but at typing-time, their adjustments will depend on what type checkers and type checker settings they are enforcing on their projects.
For example, in `mypy --strict`, a rule is in place which disallows the implicit-any for generics. So the following two equivalent formulations are treated differently:
```python
class MyAuthorizer(RenewingAuthorizer[Any]):  # ok

class MyAuthorizer(RenewingAuthorizer):  # not ok
```

I wouldn't consider it a breaking change, but it's possible that someone will need to update their types. (Typing time compatibility is a complex area; more to unpack here if you like.)

Reply:
Discussed offline; resolving this thread.